### PR TITLE
urlview: new package at 0.9-21

### PR DESCRIPTION
**Pull request by ioraff**
Read: <getkiss.org/guidestones.txt>

## Description of package
ncurses-based URL scanner, with patches from Debian as it has not had an official release in ~20 years.

## New package

- [x] Latest upstream version.
- [x] I agree to maintain this package.
- [x] I agree to notify @dylanaraps if I can no longer maintain this package.

## Existing package

- [ ] I am the maintainer of this package.

### CODE REVIEW

File: community/urlview/sources [hunk 1]
```diff
@@ -0,0 +1,2 @@
+http://ftp.debian.org/debian/pool/main/u/urlview/urlview_0.9.orig.tar.gz
+http://ftp.debian.org/debian/pool/main/u/urlview/urlview_0.9-21.diff.gz
```
Comment (dylanaraps) [inline community/urlview/sources]: Purpose of this?

File: community/urlview/depends [hunk 2]
```diff
@@ -0,0 +1,5 @@
+autoconf make
+automake make
+gzip make
+ncurses
+patch make
```
Comment (dylanaraps) [inline community/urlview/depends]: Remove this.

File: community/urlview/build [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+# The included url_handler.sh is extremely outdated -- use $BROWSER by default
+# We don't want $BROWSER to expand, so
+# shellcheck disable=SC2016
+sed -i 's|COMMAND url_handler\.sh|COMMAND \$BROWSER %s|' sample.urlview
+
+aclocal
+automake \
+    --foreign \
+    --ignore-deps \
+    --add-missing
```
Comment (dylanaraps) [inline community/urlview/build]: Why? No pre-generated files?

File: community/urlview/build [hunk 4]
```diff
@@ -0,0 +1,33 @@
+#!/bin/sh -e
+
+gunzip urlview_0.9-21.diff.gz
+patch -p1 < urlview_0.9-21.diff
+
+# Move user config to ~/.config/urlview/conf
+sed -i 's|%s\/\.urlview|%s\/\.config/urlview/conf|' urlview.c
```
Comment (dylanaraps) [inline community/urlview/build]: Don't escape forward slashes here.

File: community/urlview/depends (hunk 2 above)
Comment (ioraff) [reply community/urlview/depends]: the patch fails with `busybox patch`

Comment (dylanaraps) [reply community/urlview/depends]: Fix the patch (if it really is required).

File: community/urlview/build (hunk 3 above)
Comment (ioraff) [reply community/urlview/build]: The original sources assumes assume automake-1.11 and the debian patch is quite substantial. The build fails without this..

File: community/urlview/sources (hunk 1 above)
Comment (ioraff) [reply community/urlview/sources]: The original software hasn't been updated in nearly 20 years, but debian has been maintaining it through patches.

File: community/urlview/build (hunk 3 above)
Comment (dylanaraps) [reply community/urlview/build]: Why not just call `autoreconf` (may need `-i` or `-fis`)?

Comment (ioraff) [reply community/urlview/build]: That unfortunately isn't sufficient. I can only get it to build with these lines.

File: community/urlview/build [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    --ignore-deps \
+    --add-missing
+
+./configure \
+    --prefix=/usr
+
+make
+
+# Need to create dirs before `make install`
+install -d "$1/etc/urlview" "$1/usr/share/man/man1"
```
Comment (dylanaraps) [inline community/urlview/build]: Add a newline here.

File: community/urlview/build [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+    --foreign \
+    --ignore-deps \
+    --add-missing
+
+./configure \
+    --prefix=/usr
+
+make
+
+# Need to create dirs before `make install`
```
Comment (dylanaraps) [inline community/urlview/build]: Full stop.

File: community/urlview/build [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+
+gunzip urlview_0.9-21.diff.gz
+patch -p1 < urlview_0.9-21.diff
+
+# Move user config to ~/.config/urlview/conf
+sed -i 's|%s/\.urlview|%s/\.config/urlview/conf|' urlview.c
+sed -i 's|~/.urlview|~/.config/urlview/conf|g' urlview.man
+
+# The included url_handler.sh is extremely outdated -- use $BROWSER by default
+# We don't want $BROWSER to expand, so
```
Comment (dylanaraps) [inline community/urlview/build]: Full stops.

File: community/urlview/sources (hunk 1 above)
Comment (dylanaraps) [reply community/urlview/sources]: No https sources?

**Pull request merged**